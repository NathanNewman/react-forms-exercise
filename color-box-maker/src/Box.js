import React from 'react';

const Box = (props) => {
    const height = props.height;
    const width = props.width;
    const color = props.color;
    return (
        <div style={{ height: height, width: width, backgroundColor: color }} />
    );
}

export default Box;