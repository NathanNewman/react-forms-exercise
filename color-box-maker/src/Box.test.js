import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Box from "./Box";

// smoke test
it("renders without crashing", function() {
  render(<Box />);
});

// snapshot
test("it matches snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

describe("Box component", () => {
  it("renders a box with the specified width, height, and background color", () => {
    const { container } = render(
      <Box height={30} width={20} backgroundColor="blue"/>
    );
    const box = container.querySelector("div[style]");
    expect(box.style.height).toBe("30em");
    expect(box.style.width).toBe("20em");
    expect(box.style.backgroundColor).toBe("blue");
  });
});
