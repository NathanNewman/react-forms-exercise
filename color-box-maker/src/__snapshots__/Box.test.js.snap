// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="height: 10em; width: 10em; background-color: red;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
